sklearn/preprocessing/_polynomial.py

```diff
@@ -1150,16 +1220,19 @@ def transform(self, X):
                 )
             XBS = sparse.hstack(output_list, format="csr")
         elif self.sparse_output:
-            # TODO: Remove ones scipy 1.10 is the minimum version. See comments above.
+            # TODO: Remove once scipy 1.10 is the minimum version. See comments above.
             XBS = sparse.csr_matrix(XBS)
 
+        # replace any np.nan values with 0
+        XBS = np.nan_to_num(XBS, nan=0)
+
         if self.include_bias:
-            return XBS
+            return self._concatenate_indicator(XBS)
```

Comment:
The missingness indicators computed from the `X` passed to `.transform` (which can be a test set) should be passed as argument to `_concatenate_indicator` instead of reusing the mask extracted from the training set.